File: 디자인패턴/java 디자인 패턴/Mediator/HeatBoiler.java
public class HeatBoiler extends Participant {
    private boolean bOff = false;
    
    public HeatBoiler(Mediator mediator) {
        super(mediator);
    }

    public void On() {
        bOff = true;
        mediator.participantChange(this);
    }

    public void Off() {
        bOff = false;
        mediator.participantChange(this);
    }

    public boolean isRunning() {
        return bOff;
    }

    @Override
    public String toString() {
        if(bOff) return "# 보일러: 켜짐";
        else return "# 보일러: 꺼짐";
    }
}

File: 디자인패턴/Python 디자인 패턴/Mediator/HeatBoiler.py
from Participant import*

class HeatBoiler(Participant):
    def __init__(this, mediator: Mediator) -> None:
        super().__init__(mediator)
        this.bBoiler = False
    
    def On(this):
        this.bBoiler = True
        this.mediator.participantChange(this)
    
    def Off(this):
        this.bBoiler = False
        this.mediator.participantChange(this)
    
    def isBoiler(this):
        return this.bBoiler
    
    def strPrint(this):
        if(this.bBoiler):return "# 보일러 켜짐"
        return "# 보일러 꺼짐"
    pass